app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/EnabledMessageComposer.kt

```diff
@@ -0,0 +1,266 @@
+                                inputStateHolder.isTextExpanded
+                            ) {
+                                DropDownMentionsSuggestions(
+                                    currentSelectedLineIndex = currentSelectedLineIndex,
+                                    cursorCoordinateY = cursorCoordinateY,
+                                    membersToMention = mentionSearchResult,
+                                    searchQuery = messageComposition.value.messageText,
+                                    onMentionPicked = {
+                                        messageCompositionHolder.addMention(it)
+                                        onClearMentionSearchResult()
+                                    }
+                                )
+                            }
+                        }
+                    }
+
+                    if (inputStateHolder.showOptions) {
+                        if (additionalOptionStateHolder.additionalOptionsSubMenuState != AdditionalOptionSubMenuState.RecordAudio) {
+                            AdditionalOptionsMenu(
+                                additionalOptionsState = additionalOptionStateHolder.additionalOptionState,
+                                selectedOption = additionalOptionStateHolder.selectedOption,
+                                isEditing = messageCompositionInputStateHolder.inputType is MessageCompositionType.Editing,
+                                isSelfDeletingSettingEnabled = isSelfDeletingSettingEnabled,
+                                isSelfDeletingActive = messageComposerViewState.value.selfDeletionTimer.duration.isPositiveNotNull(),
+                                isMentionActive = messageComposerViewState.value.mentionSearchResult.isNotEmpty(),
+                                onMentionButtonClicked = {
+                                    messageCompositionHolder.startMention(
+                                        onSearchMentionQueryChanged,
+                                        onClearMentionSearchResult
+                                    )
+                                },
+                                onOnSelfDeletingOptionClicked = {
+                                    additionalOptionStateHolder.toSelfDeletingOptionsMenu()
+                                    onChangeSelfDeletionClicked()
+                                },
+                                onRichOptionButtonClicked = messageCompositionHolder::addOrRemoveMessageMarkdown,
+                                onPingOptionClicked = onPingOptionClicked,
+                                onAdditionalOptionsMenuClicked = {
+                                    if (inputStateHolder.showSubOptions) {
+                                        messageCompositionInputStateHolder.toComposing()
+                                    } else {
+                                        showAdditionalOptionsMenu()
+                                    }
+                                },
```

Comment:
question: should only this one button act like a toggle? If I get it correctly, it will close the additional options menu if the button is already selected, right? Shouldn't we do the same when user clicks on mention or rich editing second time for instance?

Reply:
Yes, and there is another ticket for that (for mention, because rich text overlaps options)